### PR TITLE
Add RTL language display to projects

### LINKED ISSUE #4621: Support displaying languages other than English
Add `lang` attributes to projects, to indicate the language to browsers and assistive technology.

Add right-to-left display styles for right-to-left languages.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Add lang attribute to project page container.
Add RTL styles for Hebrew and Arabic.
Add language files for Hebrew and Arabic

Staging branch URL: https://project-language.pfe-preview.zooniverse.org/projects/judaicadh/scribes-of-the-cairo-geniza?env=production&language=he

Fixes #4621.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Task answers are aligned left in Hebrew. I think they should be aligned right, @beckyrother?
https://project-language.pfe-preview.zooniverse.org/projects/judaicadh/scribes-of-the-cairo-geniza/classify?env=production&language=he
Screenshot showing answer text ranged left in Hebrew.

**Comment by beckyrother** [comment]
@eatyourgreens yes, agreed.

**Comment by eatyourgreens** [comment]
@beckyrother thanks! I've deleted that style from the task answers.

**Comment by beckyrother** [comment]
Those would need to be flipped as well. Also just noticed the punctuation on `? need some help with this task`

**Comment by eatyourgreens** [comment]
Yeah, those English strings need translating in the Hebrew language file.

**Comment by eatyourgreens** [comment]
The Back button warning overlaps the subject too.
Screenshot showing the Back button warning appearing above the subject photo.

**Comment by eatyourgreens** [comment]
Here are the staging links.
Arabic keyword: https://project-language.pfe-preview.zooniverse.org/projects/judaicadh/scribes-of-the-cairo-geniza/classify?reload=0&workflow=6600?env=production&language=he
Hebrew Keyword: https://project-language.pfe-preview.zooniverse.org/projects/judaicadh/scribes-of-the-cairo-geniza/classify?reload=0&workflow=6529?env=production&language=he

**Comment by eatyourgreens** [comment]
@snblickhan this PR adds Hebrew and Arabic language files for the interface text. If we can get translations for those, that would be great.

**Comment by eatyourgreens** [comment]
Also text like _0 of 1 required_ looks to be hardcoded in English at the moment, and needs to be fixed to use translations.

**Comment by eatyourgreens** [comment]
The right arrow icon on the Next button looks out-of-place in a right-to-left display, too.

**Comment by eatyourgreens** [comment]
I'm adding @srallen to this PR because I've started to edit the classifier and project nav styles.

File: app/classifier/tasks/drawing/components/DrawingToolInputIcon/DrawingToolInputIcon.jsx [hunk 1]
```diff
@@ -6,7 +6,12 @@ import icons from '../../icons';
 
 export const StyledDrawingToolInputIcon = styled.span`
   color: ${props => props.color};
-  margin-right: 1ch;
+
+  &::after {
+    content: " ";
+    margin-right: 1ch;
+    white-space: pre;
+  }
```
Comment (eatyourgreens) [inline app/classifier/tasks/drawing/components/DrawingToolInputIcon/DrawingToolInputIcon.jsx]: Would this be better fixed by just adding a space in the HTML?

File: app/pages/project/components/ProjectNavbar/components/ProjectTitle/ProjectTitle.jsx [hunk 2]
```diff
@@ -48,12 +46,14 @@ export const StyledRedirect = styled.a`
...
+  role: 'img'
 })`
   font-size: 0.75rem;
-  margin-left: 1rem;
+  margin: 0 0 0 1rem;
+
+  .rtl & {
+    margin: 0 1rem 0 0;
+  }
 `;
```
Comment (eatyourgreens) [inline app/pages/project/components/ProjectNavbar/components/ProjectTitle/ProjectTitle.jsx]: Could this also be fixed by adding a space in the HTML?

**Comment by eatyourgreens** [comment]
The Hebrew task help popup still seems to be displayed left-to-right.

**Comment by eatyourgreens** [comment]
Task help is fixed now, which involved some faffing about with the translations store and React context. That might be easier to deal with once we upgrade to React 16 (#4543.)

File: app/pages/project/components/ProjectNavbar/components/ProjectTitle/ProjectTitle.jsx (hunk 2 above)
Comment (srallen) [reply app/pages/project/components/ProjectNavbar/components/ProjectTitle/ProjectTitle.jsx]: Yeah this is a good candidate for just a space too.

**Comment by snblickhan** [comment]
Just an update that the Haifa team should be finishing up Pandora translations this week so tests will become easier. Sorry for the delay on this, many of our translators have been traveling!

**Comment by eatyourgreens** [comment]
I've added basic RTL support to the field guide and tutorial, which is probably going to break the current layout once they are translated to Hebrew.

The tutorial is hardcoded to move from left to right when going from page to page, which doesn't really work in a RTL language.

The field guide is hardcoded to slide out from the right of the view port, which also doesn't really work in an RTL layout.

**Comment by beckyrother** [comment]
Agreed, if it's possible to not include Talk that would be the best solution.

**Comment by eatyourgreens** [comment]
We could hard-code Talk to be left-to-right, and English.

**Comment by beckyrother** [comment]
Hm, just thinking: what did the old Galaxy Zoo talk do? That was in multiple RTL languages wasn't it? I'd like to follow precedent if we've dealt with this before.

**Comment by eatyourgreens** [comment]
I don't think Talk's ever been available in multiple languages. galaxyzoo.org was available in more than one language, though.

https://www.penguinwatch.org is still running, if you want to look at how Ouroboros projects worked in different languages.

**Comment by snblickhan** [comment]
@beckyrother @eatyourgreens I'd be comfortable with that at this point. We can collect feedback from the volunteers about how important it is or isn't to them to have a fully-translated Talk interface. As we've seen on other projects, people manage to create spaces for having non-English conversations even with an untranslated interface, so I think we're good to keep that going for this project.

**Comment by eatyourgreens** [comment]
Ok, Talk is now fixed as English and left-to-right text.

**Comment by eatyourgreens** [comment]
Rebased, and restaged, to bring in the language picker from #4677.

**Comment by eatyourgreens** [comment]
Rebased to the latest master and staged.

**Comment by eatyourgreens** [comment]
Ignore that. Build and deploy is broken with the latest master.
`ERROR in vendor-3420e8305b61003b59af.js from UglifyJs
Unexpected token: operator (>) [./node_modules/markdown-it-anchor/index.js:1,0][vendor-3420e8305b61003b59af.js:110133,21]`

**Comment by srallen** [comment]
I almost have a fix up for that.

**Comment by eatyourgreens** [comment]
Rebased and re-staged.

**Comment by wgranger** [comment]
Overall, the code looks pretty sound. However, the Field Guide looks a bit wonky. 

screen shot 2018-07-05 at 12 39 42 pm

Also, this is a bit pedantic, but the arrows on the tutorial are facing the opposite direction. 

screen shot 2018-07-05 at 12 41 12 pm

**Comment by eatyourgreens** [comment]
Yeah, those arrows are in the wrong font for Hebrew or Arabic. How did you get around that on Scribes? Do you load a new font when the language changes, or do you use different symbols for RTL languages?

The field guide could maybe be punted to a new PR. Depends how painful it is to rewrite the CSS.

**Comment by wgranger** [comment]
I believe on Scribes we targeted the class in the `StepThrough` component that handles those arrows and changed the CSS to flip the buttons around. Another approach is to flip the actual element, which we did with [this line](https://github.com/zooniverse/scribes-of-the-cairo-geniza/blob/a5138a500e5c93c75c23c19d80a931c77f037d69/src/components/styled/FlippedBtn.jsx#L4).

If you want to tackle these issues in another PR then I don't think there's anything else holding up this PR. I can approve/merge if you'd like to take care of the issues on a separate branch.

**Comment by eatyourgreens** [comment]
I'm inclined to say merge this and deal with the tutorial and field guide separately.

**Comment by wgranger** [comment]
Sounds good. I'll approve/merge now.

**Pull request merged**